bindings/python/example_scripts/GncNumeric.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python3
+
+##  @file
+#   @brief Some examples for GncNumeric to illustrate conversion and comparison
+#   @author Christoph Holtermann
+#   @date 2022-03-20
+#   @ingroup python_bindings_examples
+
+from gnucash import GncNumeric
+from gnucash import GNC_DENOM_AUTO, GNC_HOW_DENOM_SIGFIG, GNC_HOW_RND_ROUND
+
+# Create from int
+n1 = 1
+g1 = GncNumeric(n1)
+print("1 - create from int")
+print(n1, "(", type(n1), ") =>", g1)
+print(n1, "==", g1, ":", n1 == g1)
+print()
+
+# Create from String
+n2 = "1.1"
+g2 = GncNumeric(n2)
+print("2 - create from string")
+print(n2, "(", type(n2), ") =>", g2)
+print()
+
+# Create from two ints
+n3 = (11,10)
+g3 = GncNumeric(*n3)
+print("3 - create from two ints")
+print(n3, "(", type(n3), ") =>", g3)
+print()
+
+# Create from float
+n4 = 2.2
+g4 = GncNumeric(n4)
+print("4 - create from float")
+print(n4, "(", type(n4), ") =>", g4)
+print(n4, "==", g4, ":", n4 == g4)
+print('Same float with higher precision: %.32f' % n4)
+print()
+
+# Compare float, string and int
+print("Compare GncNumeric from float (4) to GncNumeric from string (2)")
+print(g2, "==", g4, ":", g2 == g4)
+print()
+
+print("Compare GncNumeric from float (4) to GncNumeric from int (3)")
+print(g3, "==", g4, ":", g3 == g4)
+print()
+
+print("Compare GncNumeric from string (2) to GncNumeric from int (3)")
+print(g2, "==", g3, ":", g2 == g3)
+print()
+
+# Create from float with specified precision
+print("Create from float with specified precision")
+for n in range(0,20):
+    print(GncNumeric(n4, GNC_DENOM_AUTO, GNC_HOW_DENOM_SIGFIG | n * 0x100))
+print()
+
+# Convert GncNumeric from float to remove errors
```

Comment:
see https://github.com/Gnucash/gnucash/pull/1298#pullrequestreview-915135815

Reply:
I've moved the sigfig calculation to a function like the C Makro in https://github.com/Gnucash/gnucash/blob/bbb4113a5a996dcd7bb3494e0be900b275b49a4f/libgnucash/engine/gnc-numeric.h#L218. I've removed the magic number which is identical to a bitshift by 8.

Reply:
What SWIG does with macros depends to some extent on the backend, but in general I think that it wraps them in a function that it aliases in the target language.

While it's true that GNC_HOW_DENOM_SIGFIGS is part of the gnc_numeric API I think it's a remarkably bad idea to use it for accounting. Significant Figures *does not* mean decimal places, it means the number of digits in the numerator: 2.25 has 3 significant digits, 42.25 has 4. So does 4225, but 4225.00 has 6. It should be immediately apparent how a significant digits round can create an imbalance in an accounting computation.

All of which aside you're just replaced one magic number with another. In programming a "magic number" is an undocumented numeric literal; the solution is to declare a descriptively named constant, e.g. 
```C
GNC_NUMERIC_SIGFIGS_OFFSET = 8;
/* ... */
return GNC_HOW_DENOM_SIGFIG | (n << GNC_NUMERIC_SIGFIGS_OFFSETe & GNC_NUMERIC_SIGFIGS_MASK);
```